core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -110,13 +110,18 @@ public void initialize(String name, Map<String, String> properties) {
 
     this.catalogName = name;
     this.warehouseLocation = LocationUtil.stripTrailingSlash(inputWarehouseLocation);
+
+    if (conf == null) {
```

Comment:
sorry I just saw this change now. Why do we need this? I think this should be reverted

Reply:
@nastra If Hadoop Configuration is not set then subsequent would have NullPointer Excpetion. While adding the Tests I noticed we don't have conf check in `HadoopCatalog` the way we have it in `HiveCatalog`

I can explicitly initialize conf in my Tests if you think this check is not required here. Let me know.

Reply:
`conf` always needs to be set explicitly. This is because the `HadoopCatalog` (and a few others) implement `Configurable`